### PR TITLE
correct text: pointer can point to allocatable

### PR DESCRIPTION
**Pull request by kweide**
Minor text change to eliminate wrong statement.

### CODE REVIEW

File: documentation/FortranForCProgrammers.md [hunk 1]
```diff
@@ -260,8 +260,7 @@ little like C++17's `std::visit()` on a discriminated union.
 Pointers
 --------
 Pointers are objects in Fortran, not data types.
-Pointers can point to data, arrays, and subprograms, but not to other pointers
-or to an allocatable.
+Pointers can point to data, arrays, and subprograms, but not to other pointers.
```
Comment (klausler) [inline documentation/FortranForCProgrammers.md]: What was intended by the original text is that one cannot construct a pointer in Fortran that points to the *implementation* of an allocatable object as such, as opposed to the data held within the allocatable.  One can certainly have a pointer that points to those contents, so long as the allocatable has the TARGET attribute, just as with any other data object.

**Comment by kweide** [comment]
How about:

  Pointers can point to data, arrays, and subprograms, but not to other pointers. A pointer to an `ALLOCATABLE` points to to the `ALLOCATABLE's` data.

**Comment by klausler** [comment]
That's great (except that `points to to` should be `points to`).

**Comment by kweide** [comment]
indeed,

**Comment by klausler** [comment]
Please retain the sense of the original text; pointers can point to the data managed by an allocatable, but cannot point to the allocatable itself (i.e., one cannot deallocate or reallocate an allocatable by means of a pointer to it, nor do the automatic reallocation semantics apply when an assignment is made to an allocatable through an intervening pointer).

**Comment by kweide** [comment]
Please reformulated as you see fit. Maybe you should jut copy the text in the parentheses of your last comment.

It seems that you want to address some false expectations that C or C++ programmers may have. Your are probably in a better position to address those than I, since I am more familiar with Fortran than C or C++ and thus do not share those expectation, and even find them somewhat hard to understand.

I just want to point out that the sentence "Pointers cannot point to an allocatable" is a false statement, if the terms in that sentence are taken to have their Fortran meanings.

**Comment by klausler** [comment]
A PR is out for review that extends and corrects the text.

**Comment by sscalpone** [comment]
Hi @kweide, does #464 address your concerns?

**Comment by kweide** [comment]
Yes, looks fine to me.

**Pull request closed without merging**